test/cloudtest/test_k8s_node_recovery.py

```diff
@@ -0,0 +1,387 @@
+    stalled_index = get_current_counter_index(mz)
+
+    # expect live data to be delivered at most after two minutes in production (or 300s in k8s)
+    validate_state(
+        mz,
+        stalled_index,
+        must_exceed_reached_index=True,
+        timeout_in_sec=330,
+        expected_state="live data due to node recovery",
+    )
+
+    recovered_index = get_current_counter_index(mz)
+
+    mz.revive_suspended_k8s_node(suspended_node_name)
+
+    validate_state(
+        mz,
+        recovered_index + REQUIRED_PROGRESS,
+        must_exceed_reached_index=True,
+        timeout_in_sec=10,
+        expected_state="no issues after node recovery",
+    )
+
+
+# @pytest.mark.long
+def test_unreplicated_compute_cluster_on_failing_node(
+    mz: MaterializeApplication,
+) -> None:
+    """
+    An unreplicated compute cluster is on the failed node. Queries of indexes on the compute cluster should fail, but
+    resolve within a minute or two.
+    """
+    compute_cluster = default_compute_cluster()
+    storage_cluster = default_storage_cluster()
+
+    populate(mz, compute_cluster, storage_cluster)
+    reached_index = get_current_counter_index(mz)
+
+    suspended_node_name = suspend_node_of_replica(
+        mz, compute_cluster, compute_cluster.replica_definitions[0]
+    )
+
+    # expect (live) data to be delivered after at most after two minutes in production (or 300s in k8s)
+    validate_state(
+        mz,
+        reached_index + REQUIRED_PROGRESS,
+        must_exceed_reached_index=True,
+        timeout_in_sec=330,
+        expected_state="node recovery and live data",
+    )
+
+    recovered_index = get_current_counter_index(mz)
+
+    mz.revive_suspended_k8s_node(suspended_node_name)
+
+    validate_state(
+        mz,
+        recovered_index + REQUIRED_PROGRESS,
+        must_exceed_reached_index=True,
+        timeout_in_sec=10,
+        expected_state="no issues after node recovery",
+    )
+
+
+# @pytest.mark.long
+def test_replicated_compute_cluster_on_failing_node(mz: MaterializeApplication) -> None:
+    """
+    A replicated compute cluster is on the failed node. Queries of indexes on the compute cluster should experience no
+    disruption in latency, thanks to the second replica.
+    """
+    compute_cluster = default_compute_cluster()
+    compute_cluster.replica_definitions.append(
+        ReplicaDefinition(compute_cluster.name, index=2, availability_zone="2")
+    )
+    storage_cluster = default_storage_cluster()
+
+    populate(mz, compute_cluster, storage_cluster)
+    reached_index = get_current_counter_index(mz)
+
+    suspended_node_name = suspend_node_of_replica(
+        mz, compute_cluster, compute_cluster.replica_definitions[1]
+    )
+
+    validate_state(
+        mz,
+        reached_index + REQUIRED_PROGRESS,
+        must_exceed_reached_index=True,
+        timeout_in_sec=30,
+        expected_state="live data without disruption in latency",
```

Comment:
I am getting the following failure here:
```
FAILED test_k8s_node_recovery.py::test_replicated_compute_cluster_on_failing_node - RuntimeError: Failed to achieve 'live data without disruption in latency' within 30s! Command '['kubectl', '--context', 'kind-mzcloud', '--namespace', 'default', 'exec',...
```

The reason for this is that ` suspend_node_of_replica(` above will cause 1 KinD node to be suspended, but this node can host **both** the compute replica we want to mess with **and** the storage cluster replica.

If the storage replica is borked, `SELECT` queries against both the `mv` and the `source` are going to hang permanently until the source has recovered by moving it o another K8s node, which takes more than 30 seconds.

The solution here is to run the `validate_state` under the SERIALIZABLE transaction level.